### ISSUE TITLE
How to send grant_type through the NbPasswordAuthStrategy?

### ISSUE THREAD

**Issue by chhatrapati** [akveo/nebular#566]

**Comment by alain-charles** [comment]
@chhatrapati grant-type is a parameter of the NbOAuth2AuthStrategy and not one of NbPasswordAuthStrategy.

If you want to authenticate against a Oauth2 server with a login and a password, you have to use the NbOAuth2AuthStrategy, posting the request to the token endpoint, with these options 
```
token: {
            endpoint: 'token',
            grantType: NbOAuth2GrantType.PASSWORD,
            class: NbAuthOAuth2Token,
          },
```
You can have a look at the playground/oauth2-password/ directory in the playground for an example.

Hope this helps.

**Comment by schuettecarsten** [comment]
@alain-charles You're right, but it would be a good idea to add the grant_type to the NbPasswordAuthStrategy, also. For example, our existing implementations require a grant_type value and won't work with NbPasswordAuthStrategy at the moment.

**Comment by alain-charles** [comment]
@chhatrapati if your auth server needs this value, did you try adding 
`?grant-type=yourgrantypevalue` 
 to your login endpoint ?
(not tested)

**Comment by chhatrapati** [comment]
@alain-charles i need to send grant type with username and password.

**Comment by schuettecarsten** [comment]
@alain-charles This does not work on POST methods.

**Comment by nnixaa** [comment]
So to consolidate - ideally it should be possible to add additional arbitrary parameters to `NbPasswordAuthStrategy` methods.

Currently, if you have a custom login component, the `authenticate` method accepts a data object which is passed directly to `http.request`, so there is nothing blocking here as you can add anothing you need to.

But if you want to use the out-of-the-box components we probably could introduce a new option within the method options object https://github.com/akveo/nebular/blob/master/src/framework/auth/strategies/password/password-strategy-options.ts#L44. But in this case, these options are static and there is no way to add some logic or calculate that values based on the current runtime environment.

So my conclusion would be to use [Angular HTTP Interceptors](https://angular.io/guide/http#intercepting-requests-and-responses). This excellently resolves the issue and covers all the cases above with no need to introduce any logic into Nebular Auth.

**Comment by danglephd** [comment]
Start to research about ngxadmin from 14-Sept-2020. 
I faced on 3 problems when make the login page:
- Implement LoginComponent from **NbLoginComponent**
- Add stratergy in core.module.ts : **NbPasswordAuthStrategy**
- Add AuthInterceptor implement **HttpInterceptor**, for : 
+ set Authorization on Header, set
+ set Query String params on URL, and in here I add grant-type:
```
setParams: {'password': req.body.password, 'username': req.body.username, 'grant_type': "password"},
                            setHeaders: {
                                [this.headerName]: 'Basic ' + clientCredentials,
                            },
```
And resolved all.
Now, I can get token from BE already, but the login page still show:

> 
> Oh snap!
> 
> Something went wrong.

Can you help me out?
